src/app/modules/shared_models/section_details.nim

```diff
@@ -17,25 +17,50 @@ QtObject:
     new(result, delete)
     result.setup
 
-  proc setActiveSectionData*(self: SectionDetails, item: SectionItem) =
-    self.id = item.id
-    self.sectionType = item.sectionType
-    self.joined = item.joined
-
+  proc idChanged*(self: SectionDetails) {.signal.}
   proc getId*(self: SectionDetails): string {.slot.} =
     return self.id
 
   QtProperty[string] id:
     read = getId
+    notify = idChanged
 
+  proc sectionTypeChanged*(self: SectionDetails) {.signal.}
   proc getSectionType(self: SectionDetails): int {.slot.} =
     return self.sectionType.int
 
   QtProperty[int] sectionType:
     read = getSectionType
+    notify = sectionTypeChanged
 
+  proc joinedChanged*(self: SectionDetails) {.signal.}
   proc getJoined(self: SectionDetails): bool {.slot.} =
     return self.joined
 
   QtProperty[bool] joined:
-    read = getJoined
\ No newline at end of file
+    read = getJoined
+    notify = joinedChanged
+
+  proc setActiveSectionData*(self: SectionDetails, item: SectionItem) =
+    var idChanged = false
+    var joinedChanged = false
+    var sectionTypeChanged = false
+
+    if self.id != item.id:
+      self.id = item.id
+      idChanged = true
+
+    if self.joined != item.joined:
+      self.joined = item.joined
+      joinedChanged = true
+
+    if self.sectionType != item.sectionType:
+      self.sectionType = item.sectionType
+      sectionTypeChanged = true
+
+    if idChanged:
+      self.idChanged()
+    if joinedChanged:
+      self.joinedChanged()
+    if sectionTypeChanged:
+      self.sectionTypeChanged()
```

Comment:
The template mentioned above could also be used here (`template tryUpdateWithAction*(item, field, data, action: untyped): untyped`). This function becomes:
```
self.tryUpdateWithAction(id, item.id):
   self.idChanged()
self.tryUpdateWithAction(joined, item.joined):
   self.joinedChanged()
self.tryUpdateWithAction(sectionType, item.sectionType)
   self.sectionTypeChanged()
```

Reply:
I think both approaches would have exactly the same behaviour in QML no matter what we do. 🤔

Just out of curiosity, you've seen such issues with `changed` signals on variables depending one on the other?

IIRC we have two different behaviours in the QML integration:
1. QtProperty[QVariant]: In this case qt will try to call the nim `read` whenever the qml variable is assessed. In this case, yes. I can imagine such behaviour where the nim `read` is called for a variable we didn't had the time to notify the update yet.
2. QtProperty[bool] (or any basic type): In this case Qt will cache the data. So the variable in QML will have the same value even if we're updating it in nim. The cache invalidation happens in QML only when `changed` signal is emitted.

Reply:
anyways, I just wanted to point out that we could benefit from some generic `utils` in other parts of the app as well 😄